solr/core/src/java/org/apache/solr/cli/SolrProcessManager.java

```diff
@@ -162,42 +171,60 @@ private static Optional<String> commandLine(ProcessHandle ph) {
     if (!Constants.WINDOWS) {
       return ph.info().commandLine();
     } else {
-      long desiredProcessid = ph.pid();
-      try {
-        Process process =
-            new ProcessBuilder(
-                    "wmic",
-                    "process",
-                    "where",
-                    "ProcessID=" + desiredProcessid,
-                    "get",
-                    "commandline",
-                    "/format:list")
-                .redirectErrorStream(true)
-                .start();
-        try (InputStreamReader inputStreamReader =
-                new InputStreamReader(process.getInputStream(), StandardCharsets.UTF_8);
-            BufferedReader reader = new BufferedReader(inputStreamReader)) {
-          while (true) {
-            String line = reader.readLine();
-            if (line == null) {
-              return Optional.empty();
-            }
-            if (!line.startsWith("CommandLine=")) {
-              continue;
-            }
-            return Optional.of(line.substring("CommandLine=".length()));
-          }
+      return Optional.ofNullable(pidToWindowsCommandLineMap.get(ph.pid()));
+    }
+  }
+
+  /**
+   * Gets the command lines of all java processes on Windows using PowerShell.
+   *
+   * @return a map of process IDs to command lines
+   */
+  private static Map<Long, String> commandLinesWindows() {
+    try {
+      StopWatch stopWatch = StopWatch.createStarted();
+      Process process =
+          new ProcessBuilder(
+                  "powershell.exe",
+                  "-Command",
+                  "Get-CimInstance -ClassName Win32_Process | Where-Object { $_.Name -like '*java*' } | Select-Object ProcessId, CommandLine | ConvertTo-Json -Depth 1")
+              .redirectErrorStream(true)
+              .start();
+      try (InputStreamReader inputStreamReader =
+              new InputStreamReader(process.getInputStream(), StandardCharsets.UTF_8);
```

Comment:
Why not directly tell jackson to load the JSON from the `inputStreamReader`? To me it looks strange to read it line by line and concat it together. This should be the task of the JSON parser.

Reply:
...or just use `IOUtils.toString(reader)` or better `IOUtils.toString(stream, charset)`

Reply:
Yea, added onion layers, will take a stab on peeling.

Reply:
I am sure starting the process and reading/converting the output can at end be a 4-liner using Java streaming API ;-)

Reply:
P.S.: I like the idea to use JSON output. That's much better than applying regexes and line-by line parsing of arbitrary tabular output.